### PR TITLE
Feature filter period

### PR DESCRIPTION
**Pull request by philrwebdev**
To test this feature:

- Add a plant by clicking on the '+' button on the homepage.
- Log in using credentials 'crops'/'crops'.
- Fill in all entries and set periods by using the custom timeline inputs at the bottom. This should work by either mouse-clicking or by tab-navigation (shift-tab: move backwards) and pressing Enter. To test grouping of periods into active/non-active, have one or more periods include the current time interval (mid June) (and, vice versa, one or more should exclude it).
- Click 'save', periods should be displayed at the bottom of the plant details page you're routed to.
- On the home page, there is now a filter 'Current Tasks' allowing you to filter our plants that have task periods that are currently relevant.

### CODE REVIEW

File: utils/TaskPeriodUtils.js [hunk 1]
```diff
@@ -0,0 +1,70 @@
...
+  const month = months[date.getMonth() + 1];
+  const day = date.getDate();
+  const currentInterval =
+    (day < 11 ? "early " : day < 21 ? "mid " : "late ") + month;
+
+  return currentInterval;
+}
+
+export function getActiveTasksByPlant(plants, months) {
+  const date = new Date();
```
Comment (Roland-Hufnagel) [inline utils/TaskPeriodUtils.js:43]: I don't see where you use this variable.

File: utils/TaskPeriodUtils.js [hunk 2]
```diff
@@ -0,0 +1,70 @@
...
+  const month = months[date.getMonth() + 1];
+  const day = date.getDate();
+  const currentInterval =
+    (day < 11 ? "early " : day < 21 ? "mid " : "late ") + month;
+
+  return currentInterval;
+}
+
+export function getActiveTasksByPlant(plants, months) {
```
Comment (Roland-Hufnagel) [inline utils/TaskPeriodUtils.js:42]: I'm just wondering a little bit: You export the `months` array just to get it back as a parameter?

**Review by Roland-Hufnagel — APPROVED** [review]
LGTM

File: utils/TaskPeriodUtils.js (hunk 2 above)
Comment (philrwebdev) [reply utils/TaskPeriodUtils.js:42]: Right, because I'm using the months array in other places too.

File: utils/TaskPeriodUtils.js (hunk 1 above)
Comment (philrwebdev) [reply utils/TaskPeriodUtils.js:43]: Ah, that's a remnant of refactoring this into separate functions. Will clean this up.

**Comment by philrwebdev** [comment]
Thanks, Roland!

**Pull request merged**